File: DSA_in_PHP/binary_search.php
<?php 

/**
 * Search for an item in an array using binary search algorithm
 * 
 * @param array $list 
 * @param any $target 
 * @return integer||null
 */
function binarySearch(Array $list , $target)
{
   $first = 0;
   $last = count($list);
   do {
        $mid_point = floor(($first + $last) / 2);

        if($list[$mid_point] == $target)
            return $mid_point;

        if($list[$mid_point] < $target){
            $first = $mid_point + 1; 
        }
        elseif ($list[$mid_point] > $target) {
            $last = $mid_point -1;
        }
        $count++;
   } while ($first < $last);

    return Null;
}

function verify($index){
    echo is_null($index)? 'Target not found': "Target found at {$index}";
}

$numbers = [1,2,3,4,5,6,7,8,9,10];
$result = binarySearch($numbers, 2);
verify($result);

File: DSA_in_Python/binary_search.py
def binary_search(list, target):
    first = 0
    last = len(list) - 1

    while first <= last: 
        min_point = (first + last ) // 2

        if first == last:
            return min_point
        if list[min_point] < target: 
            first = min_point +1
        elif list[min_point] > target:
            last = min_point -1
    return None

def verify (index):
    if index is not None: 
        print('Target foud at index: ', index)
    
numbers = [1,2,3,4,5,6,7,8,9,10]

result = binary_search(numbers, 0)
verify(result)
        
    